transport/server.hh

```diff
@@ -105,7 +105,7 @@ struct cql_query_state {
 struct cql_server_config {
     ::timeout_config timeout_config;
     size_t max_request_size;
-    utils::updateable_value<uint32_t> max_concurrent_requests;
+    std::function<utils::updateable_value<uint32_t> ()> get_max_concurrent_requests;
```

Comment:
It's sad to call a virtual function just to read a constant from memory,

Reply:
OK, I see why it is confusing.

This function returns an updateable_value for the given shard, not just uint32_t. It's only used to initialize the new field `_max_concurrent_requests`, which itself is also an updateable value. This causes `_max_concurrent_requests` attach to the config parameter. When checking if we are above the limit of concurrent requests, we only look at the `_max_concurrent_requests` value, we don't call this std::function.

Perhaps it's better to change it to a more descriptive name like `get_max_concurrent_requests_updateable_value` (ugh)

Reply:
No, I did not spend enough effort to read it.